src/parser/symbol_manager.cpp

```diff
@@ -0,0 +1,77 @@
+/*********************                                                        */
+/*! \file symbol_manager.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief The symbol manager
+ **/
+
+#include "parser/symbol_manager.h"
+
+namespace CVC4 {
+namespace parser {
+
+SymbolManager::SymbolManager(api::Solver* s) : d_solver(s) {}
+
+SymbolTable* SymbolManager::getSymbolTable() { return &d_symtabAllocated; }
+
+bool SymbolManager::setExpressionName(api::Term t,
+                                      const std::string& name,
+                                      bool isAssertion)
+{
+  if (d_names.find(t) != d_names.end())
+  {
+    // already named assertion, or we are not naming an assertion
+    if (!isAssertion || d_namedAsserts.find(t) != d_namedAsserts.end())
+    {
+      return false;
+    }
+  }
+  d_namedAsserts.insert(t);
+  d_names[t] = name;
+  return true;
```

Comment:
I think you need to guard against the case where `t` is not already named and `isAssertion` is `false`.

Reply:
Will fix.